tests/unit/lib/Any_unittest.cc

```diff
@@ -691,3 +691,65 @@ TEST(Any, lazy_cloneable_visitable)
 	EXPECT_FALSE(any.visitable());
 	EXPECT_THROW(any.visit(nullptr), std::logic_error);
 }
+
+TEST(AnyParameterProperties, old_api_default)
+{
+	AnyParameterProperties params = AnyParameterProperties();
+
+	EXPECT_EQ(
+	    params.get_model_selection(),
+	    EModelSelectionAvailability::MS_NOT_AVAILABLE);
+	EXPECT_EQ(
+	    params.get_gradient(), EGradientAvailability::GRADIENT_NOT_AVAILABLE);
+	EXPECT_FALSE(params.get_model());
+}
+
+TEST(AnyParameterProperties, new_api_default)
+{
+	AnyParameterProperties params = AnyParameterProperties();
+
+	EXPECT_TRUE(params.equals(ParameterProperties()));
```

Comment:
oh, and also, actually it could be `operator==` instead of equals, or?

Reply:
mmh couldnt this even be added to `bitmask_operators` so that `==` is available for all bitmasks that are defined using this?

Reply:
but wouldn't that imply that all attributes would have to be equal, i.e. matching descriptions?

Reply:
No I mean put an `operator=` in the enum class, the one defined via `enableEnumClassBitmask(ParameterProperties);`, not the AnyParameterProperties. Or doesnt that make sense?

Reply:
yes, it does, it is actually better since it then uses the underlying type of the enum to do the `static_cast`.

Reply:
but should still have the `equals` method in `AnyParameterProperties` that then uses the `operator==`, no?

Reply:
Because you still need a method to access it no? Unless you want to have a getter for `m_attribute_mask` in `AnyParameterProperties` and then compare the returned value in the code to something. Wouldn't the `equals` method be cleaner?

Reply:
see my other comment

Reply:
(I agree we need a method), but equals not maybe not the best name.
What about a getter instead? Then the caller can just use `==`